File: 0011-container-with-most-water/0011-container-with-most-water.java
import java.util.*;

class Solution {
    public int maxArea(int[] height) {
        int max = -1;
        int lt = 0;
        int rt = height.length - 1;
        while (lt < rt) {
            int width = rt - lt;
            int minHeight = Math.min(height[lt], height[rt]);
            System.out.println(minHeight);
            max = Math.max(max , width * minHeight);
            if (height[lt] <= height[rt]) {
                lt++;
            } else {
                rt--;
            }
        }
        return max;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxArea = 0
        n = len(height)

        l, r = 0, n - 1
        while l < r:
            h = min(height[l], height[r])
            w = r - l

            maxArea = max(maxArea, h * w)

            if height[l] < height[r]:
                l += 1
            elif height[l] > height[r]:
                r -= 1
            else:
                l += 1
                r -= 1
        return maxArea